### ISSUE TITLE
Adding RoadConditions to EnvironmentalConditions

### ISSUE THREAD

**Issue by ClemensLinnhoff** [OpenSimulationInterface/open-simulation-interface#418]
#### Describe the feature
RoadConditions in OSI are quite hidden. They are currently part of the lane classfication. For me, values like water film height or ice on the pavement are environmental conditions.

#### Describe the solution you would like
Therefore I propose to move the RoadConditions to the environmental conditions to make it more intuitive to find them. I know, that environmental conditions are supposed to be global values at the moment, which I would also question. This might be a discussion for another day, but conditions like fog can also occur locally. For the RoadCondition a lane ID parameter would fix this issue, when they are moved to the environmental conditions.

#### Describe the backwards compatibility
This would create some problems for backwards compatibiliy, so it could only be done in a major release.

**Comment by kmeids** [comment]
Since it is a property of the Lane, I don't see why it should be added to environmental conditions.
For example, you have a road with 4 lanes, but a water film only on one of them, having the road condition under environmental conditions will make it harder to describe the situation.

However, I would see that it should be moved directly under Lane and not under osi3::Lane::Classification

@ClemensLinnhoff, what do you think?

**Comment by ClemensLinnhoff** [comment]
I agree with putting the road condition directly under lane and I get the argument that it does not have to be in the environmental conditions class. However, we can continue this discussion on the wetness of surfaces other than the pavement. Would you see conditions like surface wetness also in the individual objects class, like stationary objects? The same class "RoadCondition" could be used directly for other surfaces and would just have to be renamed.
Then it would also be more intuitive, if every object (incl. lanes) had a surface conditions class.

**Comment by kmeids** [comment]
I agree!
As you have suggested this should also be included under static objects.
@clemenshabedank, it's short notice but I'll forward you the invitation to today's meeting.

**Comment by clemenshabedank** [comment]
@kmeids I guess you mean @ClemensLinnhoff

**Comment by kmeids** [comment]
@ClemensLinnhoff, Is this something that you would like to further discuss in the next SensorModeling meeting?

**Comment by ClemensLinnhoff** [comment]
Yes, I'd be happy to discuss this, but unfortunately the next meeting I could take part in would be on July 9th.

**Comment by ClemensLinnhoff** [comment]
> [@ClemensLinnhoff](https://github.com/ClemensLinnhoff) Are you still interested in this topic?

Yes, but I think this would be something for a major release, so I don't know how relevant this currently is.

**Comment by thomassedlmayer** [comment]
I mean, we could at least generally sync road conditions with ISO23150 and maybe even use a minor release to prepare for a major release by deprecating old fields and introducing new ones if really necessary?

> I know, that environmental conditions are supposed to be global values at the moment, which I would also question.

If I remember correctly, environmental conditions describe the current state at the ego vehicle position and are not a global state. E.g. fog values in OSI describe the current fog state at the ego vehicle position at the specified time. I remember a discussion where we even talked about introducing something like bounding boxes (like fog bounding box in OpenScenario) that could hold different locally applicable environmental conditions on a global level. But as always there was no real driver/use cases for the feature.

**Comment by ClemensLinnhoff** [comment]
> environmental conditions describe the current state at the ego vehicle

This does not make sense for perception sensor simulation in my opinion. You need to know what the conditions in a certain region around the ego vehicle are and not "at" the ego vehicle. So you would have to assume that these conditions persist at a certain radius (100, 200, 500 m) around the ego in order to use this information for perception sensor simulation, i.e. assume they are somewhat global.

But even that is only useful for very simple models. Because most relevant for perception are conditions that vary locally.

**Comment by ClemensLinnhoff** [comment]
Just checked the documentation again. [Here](https://opensimulationinterface.github.io/osi-antora-generator/asamosi/latest/gen/structosi3_1_1EnvironmentalConditions.html#_detailed_description) it says:

> These conditions apply locally around the host vehicle.

So kind of a mix between what we both said @thomassedlmayer. In my opinion, for the purpose of perception sensor simulation, this can still be considered as "global". It is homogeneous within the field of view of a sensor.  But what is "global" in a simulation environment anyways.

**Comment by thomassedlmayer** [comment]
Yes, OSI does not dictate a specific radius or similar. This leaves some room for interpretation which can be good and bad.

OSI's EnvironmentalConditions somehow feels like a rough ODD description. I guess it's not really intended to be used for sophisticated weather simulation. For most use cases it probably wouldn't matter where a small cloud of fog is exactly: If you're driving somewhere near it, a driving function would have to cope with any possible way and a sensor could just give any potential output for the defined weather condition.

Still, it is of course not exactly clear how to set and transition between environmental conditions if there is a highly variable weather state around the ego vehicle (e.g. momentary rain showers, small clouds of fog, wind gusts).

The same would apply to road conditions. Those could significantly vary across a single lane (lateral, longitudinal). The question is where and why to stop making it more and more detailed.

But back to the topic: I just asked because we're trying to tidy out old issues/PRs. If you want to continue working on this, we can discuss this in the sensor modeling project. Otherwise I'd like to close this issue for now.

**Comment by ClemensLinnhoff** [comment]
Okay then let's close it. I don't have a relevant use-case at the moment.
